### PR TITLE
Issue #682: Rm youtube hack, entering "youtube.com/tv" no longer reloads current page

### PR DESCRIPTION
**Pull request by mcomella**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -148,9 +148,16 @@ class BrowserFragment : IWebViewLifecycleFragment() {
...
         lifecycle.addObserver(cursor!!)
 
-        layout.browserOverlay.onNavigationEvent = onNavigationEvent
-        layout.browserOverlay.navigationStateProvider = NavigationStateProvider()
-        layout.browserOverlay.visibility = overlayVisibleCached ?: View.GONE
+        with (layout.browserOverlay) {
+            onNavigationEvent = this@BrowserFragment.onNavigationEvent
+            navigationStateProvider = NavigationStateProvider()
+            visibility = overlayVisibleCached ?: View.GONE
+            onPreSetVisibilityListener = { isVisible ->
```
Comment (liuche) [inline app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt]: I had forgotten that this was called in `setVisible`. Maybe add a comment in the declaration of this variable?

File: app/src/main/java/org/mozilla/focus/browser/cursor/CursorController.kt [hunk 2]
```diff
@@ -14,6 +14,7 @@ import kotlinx.android.synthetic.main.fragment_browser.*
 import org.mozilla.focus.architecture.NonNullObserver
 import org.mozilla.focus.browser.BrowserFragment
 import org.mozilla.focus.ext.getAccessibilityManager
+import org.mozilla.focus.ext.isVisible
```
[the remark this answers is not in the document]
Comment (mcomella) [reply app/src/main/java/org/mozilla/focus/browser/cursor/CursorController.kt]: It gets removed in a later commit.

**Comment by mcomella** [comment]
> Does this fix #682 as well as #733 ? Update the commit messages to reflect what these patches do, right now they only reference #682.

It fixes both of them. Github isn't smart about closing two issues with one commit, i.e. "Closes #xyz, #abc: ...", so I intended to just dupe one to other, if that's okay with you.

File: app/src/amazonWebview/java/org/mozilla/focus/iwebview/WebViewProvider.kt [hunk 3]
```diff
@@ -49,14 +53,39 @@ object WebViewProvider {
...
+private fun initWebview(webView: FirefoxAmazonWebView) = with (webView) {
     isVerticalScrollBarEnabled = true
     isHorizontalScrollBarEnabled = true
 
     // TODO This does not exist with the AmazonWebView
     //if (BuildConfig.DEBUG) {
     //    setWebContentsDebuggingEnabled(true);
     //}
+
+    // onFocusChangeListener isn't called for AmazonWebView (unlike Android's WebView)
```
[the remark this answers is not in the document]
Comment (mcomella) [reply app/src/amazonWebview/java/org/mozilla/focus/iwebview/WebViewProvider.kt]: Yes – I wrote an email asking where we could put this info.

File: app/src/amazonWebview/java/org/mozilla/focus/iwebview/WebViewProvider.kt [hunk 4]
```diff
@@ -49,14 +53,39 @@ object WebViewProvider {
...
+    // so we use the global listener instead.
+    viewTreeObserver.addOnGlobalFocusChangeListener { oldFocus: View?, newFocus: View? ->
+        if (!viewTreeObserver.isAlive) return@addOnGlobalFocusChangeListener
```
[the remark this answers is not in the document]
Comment (mcomella) [reply app/src/amazonWebview/java/org/mozilla/focus/iwebview/WebViewProvider.kt]: It's not an annotation: it's `return AT this method call`. This allows you to return from nested calls lambdas without using labels. I'm not sure why AS decided it was necessary here... 🤔

**Pull request merged**